src/datatypes/src/value.rs

```diff
@@ -268,6 +268,23 @@ impl Value {
         }
     }
 
+    /// Cast Value to f32. Return None if it's not castable;
+    pub fn as_f64(&self) -> Option<f64> {
```

Comment:
I'd prefer `as_f64_lossy` since it's not always precise to convert a u64/i64 to f64 
```suggestion
    pub fn as_f64_lossy(&self) -> Option<f64> {
```